app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonTypeFragment.kt

```diff
@@ -51,21 +53,34 @@ class PokemonTypeFragment : Fragment() {
 
     private fun showPokemonTypes(pokemonTypes: List<PokemonType>) {
         binding.pokemonAlltypesLoading.visibility = View.GONE
-        binding.pokemonAlltypesRecyclerView.apply {
-            layoutManager =
-                GridLayoutManager(context, resources.configuration.getColumnsByOrientation(COLUMNS_PORTRAIT, COLUMNS_LANDSCAPE))
-            adapter = PokemonTypesAdapter(pokemonTypes)
-            visibility = View.VISIBLE
+        pokemonTypes.let {
+            val pokemonTypesAdapter = PokemonTypesAdapter(pokemonTypes, this)
+            binding.pokemonAlltypesRecyclerView.apply {
+                layoutManager =
+                    GridLayoutManager(context, resources.configuration.getColumnsByOrientation(COLUMNS_PORTRAIT, COLUMNS_LANDSCAPE))
+                adapter = pokemonTypesAdapter
+                visibility = View.VISIBLE
+            }
         }
     }
 
     private fun showPokemonTypesError(error: String?) {
         binding.pokemonAlltypesLoading.visibility = View.GONE
-        error?.let { Toast.makeText(context, it, Toast.LENGTH_SHORT).show() }
+        Toast.makeText(context, error, Toast.LENGTH_SHORT).show()
     }
 
     companion object {
+        private const val TYPE = "types"
         private const val COLUMNS_PORTRAIT = 4
         private const val COLUMNS_LANDSCAPE = 6
     }
+
+    override fun onPokemonTypeSelected(typeSelected: PokemonType) {
```

Comment:
this should call the viewModel to set the new state of this liveData.. then we will call the startActivity

Reply:
refactored and updated the unit tests to check the functionality, thanks for the explanation